CRM/Upgrade/Incremental/sql/4.7.18.mysql.tpl

```diff
@@ -68,3 +68,12 @@ UPDATE `civicrm_state_province` SET `name`='Uttarakhand', `abbreviation`='UT' WH
 UPDATE `civicrm_state_province` SET `name`='Yunlin County' WHERE `name` = 'Yunlin Conuty';
 UPDATE `civicrm_country` SET `name`='Palestine, State of' WHERE `name` = 'Palestinian Territory';
 UPDATE `civicrm_country` SET `name`='Virgin Islands, British' WHERE `name` = 'Virgin Islands,British';
+
+-- CRM-19464 renumber address options to add in 'Supplemental Address 3'
+SELECT @option_group_id_adOpt := max(id) from civicrm_option_group where name = 'address_options';
+UPDATE civicrm_option_value SET value = value + 1, weight = weight + 1 WHERE option_group_id = @option_group_id_adOpt AND value > 3;
+INSERT INTO
+ `civicrm_option_value` (`option_group_id`, {localize field='label'}label{/localize}, `value`, `name`, `grouping`, `filter`, `is_default`, `weight`, `description`, `is_optgroup`, `is_reserved`, `is_active`, `component_id`, `visibility_id`, `icon`)
+VALUES
+  (@option_group_id_adOpt, {localize}'{ts escape="sql"}Supplemental Address 3{/ts}'{/localize}, 4, 'supplemental_address_3', NULL, 0, NULL, 4, NULL, 0, 0, 1, NULL, NULL, NULL);
```

Comment:
I've eventually got round to digging into this a bit further.  The id of the new value doesn't matter, but the weight does so that it displays in the right order.  ``ensureOptionValueExists`` is a wrapper around ``civicrm_api3('OptionValue', 'create')``  If no weight is specified a default weight at the end of the group is assigned, otherwise the specified weight value is used even if it results in duplicates.

If OptionValues are created/reordered with the GUI, the weights of all OptionValues are adjusted so that there are no duplicates, but doing this via the API can result in duplicate weights.  There is ``CRM_Core_OptionValue::addOptionValue()`` which has more logic around weights which maybe should be called instead from the API or from ``ensureOptionValueExists`` but that is straying a long way from the focus of this PR.

So we can use ``ensureOptionValueExists`` but it will result in a duplicate weight whereas the SQL I had increments whatever other existing weights there are to make space for the new one.

More generally, with incremental upgrades, how closely are we trying to make an upgraded system match a freshly installed system?  So in this case, it makes sense to put supplemental_3 after supplemental_2 in ``xml/templates/civicrm_data.tpl`` for new systems.  For an upgrade, we can try to match that (the approach I was following), or we can add it to the end of the table as ``ensureOptionValueExists`` would do. 

I'll change the logic to look for supplemental_2 and set the new weight based on that, but given that the weight is important, how do we proceed?